bool leftLine
bool midLine
bool rightLine
